File: src/main/java/org/jukeboxmc/block/BlockTnt.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemTnt;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockTnt extends Block {

    public BlockTnt() {
        super( "minecraft:tnt" );
    }

    @Override
    public ItemTnt toItem() {
        return new ItemTnt();
    }

    @Override
    public BlockType getType() {
        return BlockType.TNT;
    }

    public void setExplode( boolean value ) {
        this.setState( "explode_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isExplode() {
        return this.stateExists( "explode_bit" ) && this.getByteState( "explode_bit" ) == 1;
    }

    public void setAllowUnderwater( boolean value ) {
        this.setState( "allow_underwater_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isAllowUnderwater() {
        return this.stateExists( "allow_underwater_bit" ) && this.getByteState( "allow_underwater_bit" ) == 1;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockTnt.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.Tnt
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.extensions.toByte

class BlockTnt(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), Tnt {

   override fun isExplode(): Boolean {
       return this.getBooleanState("explode_bit")
   }

   override fun setExplode(value: Boolean): Tnt {
       return this.setState("explode_bit", value.toByte())
   }

   override fun isAllowUnderwater(): Boolean {
       return this.getBooleanState("allow_underwater_bit")
   }

   override fun setAllowUnderwater(value: Boolean): Tnt {
       return this.setState("allow_underwater_bit", value.toByte())
   }
}